src/Sylius/Bundle/ApiBundle/Validator/Constraints/OrderProductEligibilityValidator.php

```diff
@@ -57,6 +62,11 @@ public function validate(mixed $value, Constraint $constraint): void
                     $constraint->message,
                     ['%productName%' => $orderItem->getProduct()->getName()],
                 );
+            } elseif (!$orderItem->getProduct()->hasChannel($channel)) {
+                $this->context->addViolation(
+                    $constraint->message,
+                    ['%productName%' => $orderItem->getProduct()->getName()],
+                );
             }
         }
```

Comment:
I'd suggest splitting these two cases and informing the client precisely why they can't finalize the checkout:
```suggestion
        foreach ($orderItems as $orderItem) {
            $variant = $orderItem->getVariant();
            $product = $orderItem->getProduct();

            if ($variant == null || $product === null) {
                continue;
            }

            if (!$variant->isEnabled() || !$product->isEnabled()) {
                $this->context->addViolation(
                    $constraint->message,
                    ['%productName%' => $product->getName()],
                );
                continue;
            }

            if (!$product->hasChannel($channel)) {
                $this->context->addViolation(
                    $constraint->productNotInChannelMessage,
                    ['%productName%' => $product->getName()],
                );
            }
        }
```

Reply:
Fixed 👍🏻